src/illumidesk/setup_course/course.py

```diff
@@ -183,6 +190,13 @@ def run(self):
         base_url = os.environ.get('JUPYTERHUB_BASE_URL') or ''
         logger.debug('Grader container JUPYTERHUB_API_URL set to %s' % jupyterhub_api_url)
         logger.debug('Grader container JUPYTERHUB_API_TOKEN set to %s' % jupyterhub_api_token)
+        # set initial volumes dict 
+        docker_volumes = {
+            str(self.grader_root): {'bind': f'/home/{self.grader_name}'},
+            str(self.exchange_root): {'bind': '/srv/nbgrader/exchange'},
+        }
+        if self.is_shared_folder_enabled:
+            docker_volumes[str(self.grader_shared_folder)] = {'bind': f'/home/{self.grader_name}/shared'}
         self.client.containers.run(
```

Comment:
Did you evaluate the option of having the course's shared folder located in `Path(os.environ.get('MNT_ROOT'), org, 'shared', self.course_id)` vs `Path(os.environ.get('MNT_ROOT'), org, 'self.course_id', 'shared')`? My concern is having the `shared` folder as the root of the `org` path since that would mean the course folders are duplicated, for example, in `/{mnt_root}/{org}/grader-{course_id}/{course_id}` directory and in the `/{mnt_root}/{org}/shared/{course_id}` directory.

What would you think of `/{mnt_root}/{org}/grader-{course_id}/{course_id}/shared`?

Reply:
Yes, I considered that point but we would be losing the auto loading of shared folder of new courses. I mean If we create a new shared folder at that level (`/{mnt_root}/{org}/grader-{course_id}/{course_id}/`) the existing learner notebook would never know about this new shared folder because its volumes were mounted when some of those new courses did not exist.